zkvm/src/lib.rs

```diff
@@ -12,6 +12,7 @@ mod prover;
 mod scalar_witness;
 mod signature;
 mod transcript;
+mod treasury;
```

Comment:
Maybe it makes sense to make treasury/accounts a separate _crate_ (`slingshot/treasury` instead of `slingshot/zkvm/src/treasury`)? This way `zkvm` crate will be solely focused on executing the VM and having a single layer API  not cluttered with any higher-level (opinionated) protocols.

Reply:
ah, yes! sorry I meant to write that out but one of my questions was how best to structure this and a separate crate makes a lot of sense to me 👍 !

Reply:
That would also help us define what's the ZkVM should and should not be, that would allow us to simplify the internals too — if we know exactly what we need to expose to accounts, treasury, blockchain, then the rest can be freely reorganized and simplified.